src/web100srv.c

```diff
@@ -1527,647 +1326,364 @@ void child_process(int parent_pipe, SSL_CTX *ssl_context, int ctlsockfd) {
+    read(signalfd, &signal_value, sizeof(char));
+  }
+  // If no new child is trying to connect, return
+  if (!FD_ISSET(listenfd, &rfd)) return NULL;
+  // Accept the connection, initialize variables, fire up the new child.
+  ctlsockfd = 0;
+  ctlsockfd = accept(listenfd, (struct sockaddr *)&cli_addr, &clilen);
+  if (ctlsockfd <= 0) return NULL;
+  // get meta test details copied into results
+  memcpy(&meta.c_addr, &cli_addr, clilen);
+  meta.family = ((struct sockaddr *)&cli_addr)->sa_family;
+
+  memset(rmt_addr, 0, sizeof(rmt_addr));
+  // get addr details based on socket info available
+  tmp_addr = I2AddrBySockFD(get_errhandle(), ctlsockfd, False);
+  addr2a(&cli_addr, rmt_addr, sizeof(rmt_addr));
+
+  rmt_host_strlen = sizeof(rmt_host);
+  memset(rmt_host, 0, rmt_host_strlen);
+  I2AddrNodeName(tmp_addr, rmt_host, &rmt_host_strlen);
+  log_println(4, "New connection received from 0x%x [%s] sockfd=%d.", tmp_addr,
+              tmpstr, ctlsockfd);
+  procstatusenum = PROCESS_STARTED;
+  proctypeenum = CONNECT_TYPE;
+  protolog_procstatus(getpid(), getCurrentTest(), proctypeenum, procstatusenum,
+                      ctlsockfd);
+  new_child = (struct ndtchild *)calloc(1, sizeof(struct ndtchild));
+  if (pipe(child_pipe) == -1) {
+    log_println(6, "pipe() failed errno=%d", errno);
+    close(ctlsockfd);
+    free(new_child);
+    return NULL;
+  }
 
-    if (head_ptr != NULL) {
-      if ((time(0) - head_ptr->stime) > 70) {  // 70 => WAIT-TIME-THRESH
-        log_println(6,
-                    "Fault: Something in queue, but child %d (fd=%d) has "
-                    "exceeded wait time",
-                    head_ptr->pid, head_ptr->pipe);
-        // Should send new 9977 'test aborted' signal to client.  Using
-        // this for now
-        // rac 3/26/10
-
-        log_println(6, "pid=%d, client='%s', stime=%ld, qtime=%ld now=%ld",
-                    head_ptr->pid, head_ptr->addr, head_ptr->stime,
-                    head_ptr->qtime, time(0));
-        log_println(6, "pipe-fd=%d, running=%d", head_ptr->pipe,
-                    head_ptr->running);
-        // boot the client
-        child_signal = SRV_QUEUE_SERVER_BUSY;
-        write(head_ptr->pipe, &child_signal, sizeof(int));
-        close(head_ptr->pipe);
-        tpid = head_ptr->pid;
-        child_sig(-1);
-        kill(tpid, SIGTERM);
-        child_sig(tpid);
-        continue;
-      }
-    }
+  // At this point we have received a connection from a client, meaning that
+  // a test is being requested.  At this point we should apply any policy or
+  // AAA functions to the incoming connection.  If we don't like the client,
+  // we can refuse the connection and loop back to the begining.  There
+  // would need to be some additional logic installed if this AAA test
+  // relied on more than the client's IP address.  The client would also
+  // require modification to allow more credentials to be created/passed
+  // between the user and this application.
+
```

Comment:
Extraneous newline (this file doesn't seem to follow long comments with newlines)